docs/source/concept_guides/quantization.mdx

```diff
@@ -1,5 +1,256 @@
+The accumulation data type specifies the type of the result of accumulating (adding, multiplying, etc) values of the
+data type in question. For example, let's consider two `int8` values `A = 127`, `B = 127`, and let's define `C` as the
+sum of `A` and `B`:
+
+```
+C = A + B
+```
+
+Here the result is much bigger than the biggest representable value in `int8`, which is `127`. Hence the need for a larger
+precision data type to avoid a huge precision loss that would make the whole quantization process useless.
+
+
+### How do machines represent numbers?
+
+<Tip>
+
+The section is not fundamental to understand the rest. It explains in brief how numbers are represented in computers.
+Since quantization is about going from one representation to another, it can be useful to have some basics, but it is
+definitely not mandatory.
+
+</Tip>
+
+The most fundamental unit of representation for computers is the bit. Everything in computers is represented as a
+sequence of bits, including numbers. But the representation varies whether the numbers in question are integers or
+real numbers.
+
+#### Integer representation
+
+Integers are usually represented with the following bit lengths: `8`, `16`, `32`, `64`. When representing integers, two cases
+are considered:
+
+1. Unsigned (positive) integers: they are simply represented as a sequence of bits. Each bit corresponds to a power
+of two (from `0` to `n-1` where `n` is the bit-length), and the resulting number is the sum of those powers of two.
+
+Example: `19` is represented as an unsigned int8 as `00010011` because :
+```
+19 = 0 x 2^7 + 0 x 2^6 + 0 x 2^5 + 1 x 2^4 + 0 x 2^3 + 0 x 2^2 + 1 x 2^1 + 1 x 2^0
+```
+
+2. Signed integers: it is less straightforward to represent signed integers, and multiple approachs exist, the most
+common being the *two's complement*. For more information, you can check the
+[Wikipedia page](https://en.wikipedia.org/wiki/Signed_number_representations) on the subject.
+
+#### Real numbers representation
+
+Real numbers are usually represented with the following bit lengths: `16`, `32`, `64`.
+The two main ways of representing real numbers are:
+
+1. Fixed-point: there are fixed number of digits reserved for representing the integer part and the fractional part.
+2. Floating-point: the number of digits for representing the integer and the fractional parts can vary.
+
+The floating-point representation can represent bigger ranges of values, and this is the one we will be focusing on
+since it is the most commonly used. There are three components in the floating-point representation:
+
+1. The sign bit: this is the bit specifying the sign of the number.
+2. The exponent part
+  - 5 bits in `float16`
+  - 8 bits in `bfloat16`
+  - 8 bits in `float32`
+  - 11 bits in `float64`
+2. The mantissa
+  - 11 bits in `float16` (10 explictly stored)
+  - 8 bits in `bfloat16` (7 explicitly stored)
+  - 24 bits in `float32` (23 explictly stored)
+  - 53 bits in `float64` (52 explictly stored)
+
```

Comment:
I feel that this part could be clearer if we use directly an illustration.

Reply:
Not sure if we shall add fp8 as there are two formats could be too complicated here.